File: sunkatsu_mobile/lib/models/chat_message.dart
class ChatMessage {
  final String id;
  final String chatId;
  final String senderId;
  final String recipientId;
  final String? content;
  final String? imageUrl;
  final String? senderType;
  final String? timestamp;

  ChatMessage({
    required this.id,
    required this.chatId,
    required this.senderId,
    required this.recipientId,
    this.content,
    this.imageUrl,
    this.senderType,
    this.timestamp,
  });

  factory ChatMessage.fromJson(Map<String, dynamic> json) {
    return ChatMessage(
      id: json['id']?.toString() ?? '',
      chatId: json['chatId']?.toString() ?? '',
      senderId: json['senderId']?.toString() ?? '',
      recipientId: json['recipientId']?.toString() ?? '',
      content: json['content']?.toString(),
      imageUrl: json['imageUrl']?.toString(),
      senderType: json['senderType']?.toString(),
      timestamp: json['timestamp'] is String
          ? DateTime.tryParse(json['timestamp'])?.toIso8601String()
          : null,
    );
  }
}

File: backend/src/main/java/com/sunkatsu/backend/models/ChatMessage.java
package com.sunkatsu.backend.models;

import java.util.Date;

import org.springframework.data.annotation.Id;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ChatMessage {
    @Id
    private String id;
    private String chatId;
    private String senderId;
    private String recipientId;
    private String imageUrl;
    private String content;
    private Date timestamp;
    private String senderType; 
}
